pylint/checkers/utils.py

```diff
@@ -2193,6 +2193,60 @@ def is_class_attr(name: str, klass: nodes.ClassDef) -> bool:
         return False
 
 
+def is_defined(name: str, node: nodes.NodeNG) -> bool:
+    """Checks whether a node defines the given variable name."""
+    is_defined_so_far = False
+
+    if isinstance(node, nodes.NamedExpr) and node.target.name == name:
+        return True
+
+    if isinstance(node, (nodes.Import, nodes.ImportFrom)) and any(
+        node_name[0] == name for node_name in node.names
+    ):
+        return True
+
+    if isinstance(node, nodes.With):
+        is_defined_so_far = any(
```

Comment:
For my own understanding, is there a reason to switch from the `return True` pattern?

Reply:
For checks that do not `return True` immediately, there may be child nodes that define the given name. That is why `is_defined_so_far` is declared, whereby if it is False, recursively search for a child node that defines the given name.

Taking `nodes.With` as an example:
1. is_defined_so_far is True if an opened file is assigned the given name. `with open(...) as ...`
2. If the opened file is not assigned the given name, search the statements within the `with` block.

Reply:
Ah, I didn't look closely at the end of the function. Thanks, got it.